pkg/kubelet/kuberuntime/labels.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"fmt"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/kubelet/util/format"
+	kubetypes "k8s.io/kubernetes/pkg/types"
+)
+
+const (
+	kubernetesPodDeletionGracePeriodLabel    = "io.kubernetes.pod.deletionGracePeriod"
+	kubernetesPodTerminationGracePeriodLabel = "io.kubernetes.pod.terminationGracePeriod"
+
+	kubernetesContainerHashLabel                   = "io.kubernetes.container.hash"
+	kubernetesContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
+	kubernetesContainerTerminationMessagePathLabel = "io.kubernetes.container.terminationMessagePath"
+	kubernetesContainerPreStopHandlerLabel         = "io.kubernetes.container.preStopHandler"
+)
+
+type labeledPodSandboxInfo struct {
+	PodName      string
+	PodNamespace string
+	PodUID       kubetypes.UID
+}
+
+type annotatedPodSandboxInfo struct {
+	Annotations               map[string]string
+	PodDeletionGracePeriod    *int64
+	PodTerminationGracePeriod *int64
+}
+
+type labeledContainerInfo struct {
+	PodName                   string
+	PodNamespace              string
+	PodUID                    kubetypes.UID
+	PodDeletionGracePeriod    *int64
+	PodTerminationGracePeriod *int64
+	Name                      string
+	Hash                      uint64
+	RestartCount              int
+	TerminationMessagePath    string
+	PreStopHandler            *api.Handler
+}
+
+// newPodLabels creates pod labels from api.Pod.
+func newPodLabels(pod *api.Pod) map[string]string {
+	labels := map[string]string{}
+	for k, v := range pod.Labels {
+		labels[k] = v
```

Comment:
@yujuhong Do we want to store pod labels and annotations on sandbox?

At least we don't need them for now.